src/Orleans/Core/ClusterClient.cs

```diff
@@ -118,26 +126,31 @@ public void Abort()
 
         private async Task Stop(bool gracefully)
         {
-            if (this.state == LifecycleState.Disposed) return;
-
+            if (this.IsDisposing) return;
             using (await this.initLock.LockAsync().ConfigureAwait(false))
             {
                 if (this.state == LifecycleState.Disposed) return;
-                if (gracefully)
+                try
```

Comment:
`state` could never be `Disposing` here: only `Disposed` or `Invalid`. I'm not entirely sure if the state can ever actually transition to `Invalid` given `SafeExecute`, but maybe through some strange turn of events